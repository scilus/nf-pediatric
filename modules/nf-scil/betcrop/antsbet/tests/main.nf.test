nextflow_process {

    name "Test Process BETCROP_ANTSBET"
    script "../main.nf"
    process "BETCROP_ANTSBET"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "betcrop"
    tag "betcrop/antsbet"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    test("betcrop - antsbet") {

        setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-scil/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "antsbet.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/t1_unaligned.nii.gz"),
                        file("\${test_data_directory}/t1_template.nii.gz"),
                        file("\${test_data_directory}/t1_brain_probability_map.nii.gz")
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
