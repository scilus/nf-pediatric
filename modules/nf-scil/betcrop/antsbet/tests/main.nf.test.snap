{
    "betcrop - antsbet": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__t1_bet.nii.gz:md5,31ccfe501376952248b4d933645f1304"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__t1_bet_mask.nii.gz:md5,0874069d069f06bfc40d7f6be2347259"
                    ]
                ],
                "2": [
                    "versions.yml:md5,e7d67d24efb0a133d53537fed65516de"
                ],
                "mask": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__t1_bet_mask.nii.gz:md5,0874069d069f06bfc40d7f6be2347259"
                    ]
                ],
                "t1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__t1_bet.nii.gz:md5,31ccfe501376952248b4d933645f1304"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e7d67d24efb0a133d53537fed65516de"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-23T16:07:14.789765"
    }
}